azure-mgmt-graph-rbac/src/main/java/com/azure/management/graphrbac/implementation/ServicePrincipalsImpl.java

```diff
@@ -98,10 +98,14 @@ public ServicePrincipal getByName(String spn) {
                     ));
                 })
                 .map(result -> {
-                    if (result == null || result.toIterable() == null || !result.toIterable().iterator().hasNext()) {
+                    if (result == null) {
```

Comment:
Again, I think reactor does not allow `null` at all. You need to use `map` and `switchIfEmpty`.